src/components/ha-target-picker.ts

```diff
@@ -253,7 +253,9 @@ export class HaTargetPicker extends SubscribeMixin(LitElement) {
     await this.updateComplete;
     setTimeout(() => {
       this._inputElement?.open();
```

Comment:
Instead of waiting on two paints, try:
1. `await nextRender()` (instead of `updateComplete`)
2. Call focus first
3. Then call open

Reply:
Hmm, calling focus first doesn't work for me. I do like the `await nextRender()` better than `setTimeout` though, so I've incorporated `await nextRender()` into my original code. Let me know if you think I missed something with focus first.

Reply:
Yeah, the individual pickers and `ha-combo-box` should really return promises on focus, so you could just await it.  Right now the open and focus just create competing chains.

Regardless, I would still call focus first to prevent any scrolling issues, or are you saying one paint is not enough time?

Reply:
@steverep If I place `focus` before `show`, the focus doesn't happen, seemingly regardless of any updateComplete or setTimeouts. It's looking like focus is a no-op on a hidden element.  And I agree, promises would be nice, but `show` doesn't seem to return a promise, I was originally hoping for that as well.